app/src/main/java/com/github/sdp_begreen/begreen/map/Bin.kt

```diff
@@ -0,0 +1,28 @@
+package com.github.sdp_begreen.begreen.map
+
+import com.google.android.gms.maps.model.LatLng
+
+data class Bin(var id: String? = null, val type: BinType, val lat : Double, val long : Double) {
```

Comment:
I took the following approach: to be sure that those IDs are indeed unique, we need to synchronize their generation among all users. I did that by generating the ID directly on the database (when we call `addBin(bin)` and to store this ID in the `bin` given to `addBin`. That way, we are sure that Bin IDs are unique. 
Since the class `Bin` is a data class (as @AlSchlo pointed out, thanks for that), you can see [here](https://kotlinlang.org/docs/data-classes.html) that data classes automatically override `equals()` and `hashCode()` methods with the attributes in the constructor. Hope that answers your doubts.

Reply:
Thank you for your explanations